test/xnumber-test.ts

```diff
@@ -0,0 +1,73 @@
+// =============================================================================
+// Fermat.js | Extended Number Tests
+// (c) Mathigon
+// =============================================================================
+
+
+import tape from 'tape';
+import {XNumber} from '../src';
+
+
+const n = (n: number, d?: number, u?: '%'|'π') => new XNumber(n, d, u);
+const str = (s: string) => XNumber.fromString(s)?.toString();
+const dec = (s: number) => XNumber.fractionFromDecimal(s)?.toString();
+
+tape('XNumber Constructors', (test) => {
+  test.equal(n(2.5).toString(), '2.5');
+  test.equal(n(4, -2).toString(), '–4/2');
+  test.equal(n(-2, -3).toString(), '2/3');
+  test.equal(n(10, undefined, '%').toString(), '10%');
+  test.equal(n(1, 2, 'π').toString(), '1/2π');
+  test.equal(n(-1, undefined, 'π').toString(), '–π');
+  test.equal(n(0, 2, 'π').toString(), '0');
+  test.end();
+});
+
+tape('XNumber Arithmetic', (test) => {
+  test.equal(n(2).add(n(3)).toString(), '5');
+  test.equal(n(-2).add(3).toString(), '1');
+  test.equal(n(2).add(n(-3)).toString(), '–1');
+  test.equal(n(-2).add(-3).toString(), '–5');
+  // test.equal(n(2.4).add(1.2).toString(), '3.6');  floating-point error
```

Comment:
Before I noticed this comment, I was thinking, "the only thing I can think of adding to the tests would be some arithmetic using decimals like 0.2341583 * 3.847123, just to prove that the decimal–fraction conversion holds up there."